### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR adds the initial implementation of the module. 

The readme details extensively the motivation of the module, as well as the decisions made.

cc @acostalima

### CODE REVIEW

**Review by acostalima — APPROVED** [review]
LGTM 🚀

File: README.md [hunk 1]
```diff
@@ -1,6 +1,73 @@
...
+[![dependencies Status](https://david-dm.org/vasco-santos/set-delayed-interval/status.svg)](https://david-dm.org/vasco-santos/set-delayed-interval)
+[![JavaScript Style Guide](https://img.shields.io/badge/code_style-standard-brightgreen.svg)](https://standardjs.com)
 
+> An asynchronous setInterval that is properly delayed using promises and can be delayed on boot
+
+## Motivation
+
+The native implementation of `setInterval` expects synchronous functions to be executed. When asynchronous functions are provided, some unexpected behaviors might appear. This module mimics the `setInterval` native functionality with support for promises in a way that the interval timer is delayed by the execution of that promise. This results in equal stop times for each run, as soon as the promise is resolved.
+
+For some scenarios, it is important to add a boot delay before running a recurrent task. This module also supports a custom first delay before the interval start. The `clearDelayedInterval` can stop both the initial delay and the interval.
```
Comment (jacobheun) [inline README.md]:
```suggestion
For some scenarios, it is useful to add a start delay before running a recurrent task. This module also supports a custom delay before the interval starts. The `clearDelayedInterval` can stop both the initial delay and the interval.
```

**Review by jacobheun — APPROVED** [review]
minor language suggestion, otherwise lgtm

**Comment by vasco-santos** [comment]
Thanks for having a look on this @jacobheun

**Pull request merged**